internal/service/networkcontainer/resource_network_container.go

```diff
@@ -217,30 +216,36 @@ func resourceUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	containerID := ids["container_id"]
 
 	container := new(admin.CloudProviderContainer)
+	hasChange := false
 
 	if d.HasChange("atlas_cidr_block") {
+		hasChange = true
 		atlasCidrBlock := d.Get("atlas_cidr_block").(string)
 		providerName := d.Get("provider_name").(string)
 		container.AtlasCidrBlock = &atlasCidrBlock
 		container.ProviderName = &providerName
 	}
 
 	if d.HasChange("provider_name") {
+		hasChange = true
 		providerName := d.Get("provider_name").(string)
 		container.ProviderName = &providerName
 	}
 
 	if d.HasChange("region_name") {
-		regionName, _ := conversion.ValRegion(d.Get("region_name"))
-		container.RegionName = &regionName
+		hasChange = true
 	}
+	// regionName is a required field.
```

Comment:
Hi @BrettFolkins, first of all thank you so much for your contribution!

May I ask if there is a way to avoid the `hasChange` logic? Based on what you said an alternative to this is that you can put these three lines of code just inside the if statement `	if !reflect.DeepEqual(container, admin.CloudProviderContainer{}) {` and actually update the if statement as 
```
if !reflect.DeepEqual(container, admin.CloudProviderContainer{}) || d.hasChange("region_name") {
```

Am I missing something?